Sources/Sentry/SentryPerformanceTracker.m

```diff
@@ -135,7 +135,9 @@ - (void)finishSpan:(SentrySpanId *)spanId withStatus:(SentrySpanStatus)status
         [self.spans removeObjectForKey:spanId];
     }
 
-    [spanTracker finishWithStatus:status];
+    @synchronized(self.activeStack) {
+        [spanTracker finishWithStatus:status];
+    }
```

Comment:
`l`: Why this change?

Reply:
To avoid a race condition where we could be adding a child to a span being finished.